graphula-core/src/Graphula.hs

```diff
@@ -138,9 +139,42 @@ type family GraphulaContext (m :: Type -> Type) (ts :: [Type]) :: Constraint whe
    GraphulaContext m (t ': ts) = (GraphulaNode m t, GraphulaContext m ts)
 
 class EntityKeyGen a where
+  -- | Explicitly generate a key for a node
   genEntityKey :: Gen (Maybe (Key a))
   genEntityKey = pure Nothing
 
+  -- | Update the node with its generated key if applicable
+  --
+  -- This is important when generating nodes that have a generated
+  -- compound key since persistent doesn't check that the compound
+  -- key and the columns in the node are actually the same.
+  --
+  emplaceKey :: a -> Key a -> a
```

Comment:
Purely aesthetic. I'm not crazy about the name `emplace`, I also think `embed` might be more correct.